compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/PackageLoader.java

```diff
@@ -143,7 +143,7 @@ private PackageLoader(CompilerContext context) {
         this.lockEnabled = Boolean.parseBoolean(options.get(LOCK_ENABLED));
         this.repos = genRepoHierarchy(Paths.get(options.get(PROJECT_DIR)));
         this.manifest = ManifestProcessor.getInstance(context).getManifest();
-        this.lockFile = LockFileProcessor.getInstance(context).getLockFile();
+        this.lockFile = LockFileProcessor.getInstance(context, this.lockEnabled).getLockFile();
```

Comment:
Yes we had to do this change to avoid any failures that can occur when reading the Ballerina.lock file. We changed the format of the Ballerina.lock by replacing all occurrences of 'package' to 'module' and because of this we had to change the TOML parser of the Ballerina.lock file. Previously we always read the lock file regardless of whether the 'lockEnabled' flag is given or not during compile time. So now we only read the lock file only if the flag is provided.